spec/models/post_mover_spec.rb

```diff
@@ -47,16 +55,21 @@
     context 'success' do
 
       it "enqueues a job to notify users" do
-        topic.stubs(:add_moderator_post)
+        Jobs.stubs(:enqueue)
         Jobs.expects(:enqueue).with(:notify_moved_posts, post_ids: [p2.id, p4.id], moved_by_id: user.id)
         topic.move_posts(user, [p2.id, p4.id], title: "new testing topic name")
       end
 
       it "adds a moderator post at the location of the first moved post" do
-        topic.expects(:add_moderator_post).with(user, instance_of(String), has_entries(post_number: 2))
         topic.move_posts(user, [p2.id, p4.id], title: "new testing topic name")
+        expect(topic.posts).to include(a_moderator_post(post_number: 2, sort_order: 2))
```

Comment:
I don't think this assertion alone is sufficient since the test may be a false positive if `topic.posts`already contains the moderator post.